Common++/src/SystemUtils.cpp

```diff
@@ -183,18 +185,29 @@ void createCoreVectorFromCoreMask(CoreMask coreMask, std::vector<SystemCore>& re
 	}
 }
 
-std::string executeShellCommand(const std::string &command)
+namespace
 {
-	FILE* pipe = POPEN(command.c_str(), "r");
-	if (!pipe) return "ERROR";
+	struct PcloseDeleter
+	{
+		void operator()(FILE* ptr) const { PCLOSE(ptr); }
```

Comment:
better check the `ptr`
```       
 if (ptr) {
            PCLOSE(ptr);
        }
```

Reply:
The deleter is made to be used only by the smart pointer and those already skip calling the deleter on nullptr.